clean up node problem detector task page (#24772)
kubernetes/website: content/en/docs/tasks/debug-application-cluster/monitor-node-health.md

---
title: Monitor Node Health
content_type: task
reviewers:
- Random-Liu
- dchen1107
---

<!-- overview -->

*Node problem detector* is a daemon for monitoring and reporting about a node's health.

> You can run Node problem detector as a [DaemonSet](/docs/concepts/workloads/controllers/daemonset/)

Reviewer: Also valid:
```suggestion
You can run the node problem detector as a `DaemonSet`
```

> You can run node problem detector as a [DaemonSet](/docs/concepts/workloads/controllers/daemonset/)

Author: Added.

or as a standalone daemon. Node problem detector collects information about node problems from various daemons
and reports these conditions to the API server as [NodeCondition](/docs/concepts/architecture/nodes/#condition)
and [Event](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#event-v1-core).

> To learn how to install and use the Node problem detector, see the

Reviewer: I'd write:
```suggestion
To learn more about the node problem detector, see the
```

[Node problem detector project documentation](https://github.com/kubernetes/node-problem-detector).

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}}

<!-- steps -->

## Limitations

> * Node problem detector only supports file based kernel log.

Reviewer 2: nit: Here and everything title case "Node Problem Detector"

Author: OK. I can create a follow-up PR. Thanks for reviewing!

}
```

> ### Change log path

Reviewer: ```suggestion
### Configure the path for the kernel log device {#kernel-log-device-path}
```

(I think it's OK to break existing hyperlinks because there won't be many of these, and the old title could be misleading).

Author: Good idea. I'd like to shorten the heading.

Check your kernel log path location in your operating system (OS) distribution.

> The kernel log is not found in the same location in all OS distributions.

Reviewer: ```suggestion
The Linux kernel [log device](https://www.kernel.org/doc/Documentation/ABI/testing/dev-kmsg) is usually presented as `/dev/kmsg`, but may not found in the same location in all OS distributions.
```
How about this?

Author: It needs another tweak.

The `log` field in `config/kernel-monitor.json` is the log path inside the container.

> You can always configure it to match the path location in your OS distribution.

Reviewer: ```suggestion
You can configure it to match the device path as seen in by the node problem detector.
```

> ### Support other log format {#support-other-log-format}

Reviewer: Optional extra tweak
```suggestion
### Add support for another log format {#support-other-log-format}
```
